src/app/shared/related-items/related-items.component.ts

```diff
@@ -0,0 +1,239 @@
+        });
+    } else if (relationship_type === 'file-paper') {
+      this.apiFileService.getFile(this.record_id).subscribe(
+        (data: any) => {
+          this.records = data['hits']['hits'][0]['_source']['publishedArticles'];
+        });
+    } else if (relationship_type === 'dataset-specimen') {
+      this.apiFileService.getDataset(this.record_id).subscribe(
+        (data: any) => {
+          this.records = data['hits']['hits'][0]['_source']['specimen'];
+        });
+    } else if (relationship_type === 'dataset-file') {
+      this.apiFileService.getDataset(this.record_id).subscribe(
+        (data: any) => {
+          this.records = data['hits']['hits'][0]['_source']['file'];
+        });
+    } else if (relationship_type === 'dataset-paper') {
+      this.apiFileService.getDataset(this.record_id).subscribe(
+        (data: any) => {
+          this.records = data['hits']['hits'][0]['_source']['publishedArticles'];
+        });
+    } else if (relationship_type === 'dataset-analysis') {
+      this.apiFileService.getAnalysesByDataset(this.record_id).subscribe(
+        (data: any) => {
+          this.records = data['hits']['hits'];
+        });
+    } else if (relationship_type === 'organism-paper') {
+      this.apiFileService.getOrganism(this.record_id).subscribe(
+        (data: any) => {
+          this.records = data['hits']['hits'][0]['_source']['publishedArticles'];
+        });
+    } else if (relationship_type === 'organism-specimen') {
+      this.apiFileService.getOrganismsSpecimens(this.record_id).subscribe(
+        (data: any) => {
+          this.records = data['hits']['hits'];
+        });
+    } else if (relationship_type === 'specimen-paper') {
+      this.apiFileService.getSpecimen(this.record_id).subscribe(
+        (data: any) => {
+          this.records = data['hits']['hits'][0]['_source']['publishedArticles'];
+        });
+    } else if (relationship_type === 'specimen-specimen') {
+      this.apiFileService.getSpecimenRelationships(this.record_id).subscribe(
+        (data: any) => {
+          this.records = data['hits']['hits'];
+        });
+    } else if (relationship_type === 'specimen-file') {
+      this.apiFileService.getSpecimenFiles(this.record_id).subscribe(
+        (data: any) => {
+          this.records = data['hits']['hits'];
+        });
+    } else if (relationship_type === 'specimen-analysis') {
+      this.apiFileService.getAnalysesBySample(this.record_id).subscribe(
+        (data: any) => {
+          this.records = data['hits']['hits'];
+        });
+    }
+  }
+
+  // get table headers
+  get_all_fields() {
+    return setting[this.source_type][this.target_type]['all'];
+  }
+
+  get_displayed_fields() {
+    let results: string[] = [];
+    const all_fields = this.get_all_fields();
+    // use all_fields to conserve the display order
+    for (let i = 0; i < all_fields.length; i++) {
+      const curr = all_fields[i];
+      if (this.isDisplayed(curr)){
```

Comment:
Don't forget to add space between ) {, or lint will give an error

Reply:
done in 083df0d